File: scripts/processAndUploadFiles.py
import os
import glob
import requests
import dotenv

dotenv.load_dotenv()
dir_path = os.path.normpath(os.path.join(os.path.dirname(__file__), '../'))

# API URL
url = 'https://api.schickl.app/oeh.schickl.app/overwrite'
header = {'Authorization': os.getenv('AUTHORIZATION_TOKEN')}

# GET-Anfrage an die API senden
print('Read overwrite directory')
response = requests.get(url, headers=header)
if response.status_code == 200:
  print(f'- Success: {response.json()}')
else:
  print(f'- Error: {response.text}')

# DELETE-Anfrage an die API senden
response = requests.delete(url, headers=header)

for file_path in glob.glob(os.path.join(dir_path, '[a-z][a-z].*')):
  filename, file_extension = os.path.splitext(os.path.basename(file_path))
  if file_extension in ['.md', '.json']:
    filename_to_send = filename  
  else:
    filename_to_send = os.path.basename(file_path)  
  
  with open(file_path, 'r', encoding='utf-8') as file:
    data = {
      'filename': filename_to_send,
      'content': file.read()
    }

    print(f'Sending file {file_path}')
    response = requests.put(url, json=data, headers=header)

    if response.status_code == 200:
      print(f'- Success: {response.json()}')
    else:
      print(f'- Error: {response.text}')

File: scripts/processAndUploadFiles.mjs
// ECMAScript Module (ESM) Syntax
import fs from 'fs';
import path from 'path';
import _ from 'lodash';
import fetch from 'node-fetch';
import dotenv from 'dotenv';

dotenv.config();

const directoryPath = path.join(process.cwd(), 'overwrite'); // Geändert, um __dirname in ESM zu ersetzen
const result = {};

fs.readdirSync(directoryPath).forEach(file => {
    const fullPath = path.join(directoryPath, file);
    const stats = fs.statSync(fullPath);

    if (stats.isFile()) {
        const content = fs.readFileSync(fullPath, 'utf8');

        const parsed = path.parse(file);
        console.log("filename:", parsed.name, ", extension:", parsed.ext);

        let value;
        if (parsed.ext === '.json') {
            value = JSON.parse(content);
        } else {
            value = content;
        }

        _.set(result, parsed.name, value);
    } else {
        console.log(`${fullPath} ist ein Verzeichnis und wird übersprungen.`);
    }
});

const dataToSend = JSON.stringify({
    filename: 'oeh.json',
    content: JSON.stringify(result)
});

const token = process.env.AUTHORIZATION_TOKEN;

fetch('https://api.schickl.app/oeh.schickl.app/overwrite?filename=oeh.json', {
    method: 'PUT',
    body: dataToSend,
    headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${token}`
    }
})
.then(response => response.json())
.then(json => console.log(json))
.catch(err => console.error('Error:', err));

